encoding/ccf/simple_type_utils.go

```diff
@@ -0,0 +1,247 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ccf
+
+import "github.com/onflow/cadence"
+
+// IMPORTANT:
+//
+// Don't change existing simple type IDs.
+//
+// When new simple cadence.Type is added,
+// - add new ID to the end of existing IDs,
+// - add new simple cadence.Type and its ID in simpleTypeIDByType()
+
+const ( // Cadence simple type IDs
+	TypeBool = iota
+	TypeString
+	TypeCharacter
+	TypeAddress
+	TypeInt
+	TypeInt8
+	TypeInt16
+	TypeInt32
+	TypeInt64
+	TypeInt128
+	TypeInt256
+	TypeUInt
+	TypeUInt8
+	TypeUInt16
+	TypeUInt32
+	TypeUInt64
+	TypeUInt128
+	TypeUInt256
+	TypeWord8
+	TypeWord16
+	TypeWord32
+	TypeWord64
+	TypeFix64
+	TypeUFix64
+	TypePath
+	TypeCapabilityPath
+	TypeStoragePath
+	TypePublicPath
+	TypePrivatePath
+	TypeAuthAccount
+	TypePublicAccount
+	TypeAuthAccountKeys
+	TypePublicAccountKeys
+	TypeAuthAccountContracts
+	TypePublicAccountContracts
+	TypeDeployedContract
+	TypeAccountKey
+	TypeBlock
+	TypeAny
+	TypeAnyStruct
+	TypeAnyResource
+	TypeMetaType
+	TypeNever
+	TypeNumber
+	TypeSignedNumber
+	TypeInteger
+	TypeSignedInteger
+	TypeFixedPoint
+	TypeSignedFixedPoint
+	TypeBytes
+	TypeVoid
+	TypeFunction
```

Comment:
Similarly, along with `AttachmentType`, two new simple types were also introduced, namely `AnyStructAttachmentType` and `TheAnyResourceAttachmentType`. i.e:

https://github.com/onflow/cadence/blob/2e801370afbd4cc24f1e56c613257b49e97608bf/types.go#L92

and

https://github.com/onflow/cadence/blob/2e801370afbd4cc24f1e56c613257b49e97608bf/types.go#L132